src/GovUk.Education.ExploreEducationStatistics.Common.Tests/FunctionsIntegrationTest.cs

```diff
@@ -50,6 +52,57 @@ public abstract class FunctionsIntegrationTest<TFunctionsIntegrationTestFixture>
         await context.ClearTestData();
     }
 
+    // The following code was taken from this article which explains a fast way to clear all rows from Azure Table Storage
+    // https://medium.com/medialesson/deleting-all-rows-from-azure-table-storage-as-fast-as-possible-79e03937c331
+    protected async Task ClearAzureDataTableTestData(string connectionString)
+    {
+        var tableServiceClient = new TableServiceClient(connectionString);
+
+        var tables = await tableServiceClient.QueryAsync().ToListAsync();
+
+        foreach (var table in tables)
+        {
+            var tableClient = tableServiceClient.GetTableClient(table.Name);
+
+            var entities = tableClient
+                .QueryAsync<TableEntity>(select: new List<string>() { "PartitionKey", "RowKey" }, maxPerPage: 1000);
+
+            await entities.AsPages().ForEachAwaitAsync(async page => {
+                // Since we don't know how many rows the table has and the results are ordered by PartitonKey+RowKey
+                // we'll delete each page immediately and not cache the whole table in memory
+                await BatchManipulateEntities(tableClient, page.Values, TableTransactionActionType.Delete).ConfigureAwait(false);
+            });
+        }
+    }
+
+    /// <summary>
+    /// Groups entities by PartitionKey into batches of max 100 for valid transactions
+    /// </summary>
+    /// <returns>List of Azure Responses for Transactions</returns>
+    private static async Task<List<Response<IReadOnlyList<Response>>>> BatchManipulateEntities<T>(
+        TableClient tableClient, 
+        IEnumerable<T> entities, 
+        TableTransactionActionType tableTransactionActionType) where T : class, ITableEntity, new()
+    {
+        var groups = entities.GroupBy(x => x.PartitionKey);
+        var responses = new List<Response<IReadOnlyList<Response>>>();
+        foreach (var group in groups)
+        {
+            List<TableTransactionAction> actions;
+            var items = group.AsEnumerable();
+            while (items.Any())
+            {
+                var batch = items.Take(100);
+                items = items.Skip(100);
+
+                actions = [.. batch.Select(e => new TableTransactionAction(tableTransactionActionType, e))];
+                var response = await tableClient.SubmitTransactionAsync(actions).ConfigureAwait(false);
+                responses.Add(response);
+            }
+        }
+        return responses;
+    }
```

Comment:
Just discovered that there's also a method `.WithInMemoryPersistence` which we could try adding to the following code where we configure the Azurite container to enable the `--inMemoryPersistence option`.

https://github.com/dfe-analytical-services/explore-education-statistics/blob/65600c392af863af19bcfcee601ec60ff621acbb/src/GovUk.Education.ExploreEducationStatistics.Notifier.Tests/NotifierFunctionsIntegrationTest.cs#L137-L139

That might also help boost performance?

```
    private readonly AzuriteContainer _azuriteContainer = new AzuriteBuilder()
        .WithImage("mcr.microsoft.com/azure-storage/azurite:3.27.0")
        .WithInMemoryPersistence()
        .Build();
```

Reply:
Nice find!

I've added this little beauty in. I've had to upgrade our Azurite version to 3.31.0 in doing so (well, it needed to be on 3.28.0 or greater, but I took this opportunity to upgrade to the latest!). I've also fixed the other places in the code we use this TestContainer (some of them were actually inconsistent with our data-storage docker container image!)

Reply:
Cool! Glad to know this is working👍